Libraries/CMSIS/Device/Maxim/MAX32520/Include/aes_regs.h

```diff
@@ -85,7 +87,7 @@ extern "C" {
  * @ingroup aes_registers
  * Structure type to access the AES Registers.
  */
-typedef struct {
+typedef struct __attribute__((deprecated("mxc_aes_regs_t struct and aes_regs.h no longer supported. Use aeskeys_regs.h and MXC_AESKEYS (mxc_aeskeys_regs_t) for AES Key Access. 1-10-2023"))) {
```

Comment:
Consider wrapping this in an `#if defined(__GNUC__)` to avoid compatibility issues with IAR/Keil.  

@ftariqAnalogTx Can you confirm whether IAR/Keil support the deprecated attribute?

Reply:
I added the guards for IAR/Keil.

Reply:
Would something like this work instead?  Only thing that bothers me is that on IAR/Keil you have different behavior by routing to the new struct, and the new struct has different member names.

```C
#if defined(__GNUC__)
__attribute__((deprecated("mxc_aes_key_regs_t struct and aes_key_regs.h no longer supported. Use aeskeys_regs.h and MXC_AESKEYS (mxc_aeskeys_regs_t) for AES Key Access. 1-10-2023")))
#else
#warning "mxc_aes_key_regs_t struct and aes_key_regs.h no longer supported. Use aeskeys_regs.h and MXC_AESKEYS (mxc_aeskeys_regs_t) for AES Key Access. 1-10-2023"
#endif
typedef struct {
    __IO uint32_t aes_key0[4];          /**< <tt>\b 0x00:</tt> AES_KEY AES_KEY0 Register */
    __R  uint32_t rsv_0x10_0x1f[4];
    __IO uint32_t aes_key1[4];          /**< <tt>\b 0x20:</tt> AES_KEY AES_KEY1 Register */
    __R  uint32_t rsv_0x30_0xff[52];
    __IO uint32_t aes_key2[4];          /**< <tt>\b 0x100:</tt> AES_KEY AES_KEY2 Register */
    __R  uint32_t rsv_0x110_0x17f[28];
    __IO uint32_t aes_key3[4];          /**< <tt>\b 0x180:</tt> AES_KEY AES_KEY3 Register */
} mxc_aes_key_regs_t;

Reply:
You're right. That didn't come to mind when I was making the changes.